app/scripts/src/main/scala/org/bitcoins/scripts/ScanBitcoind.scala

```diff
@@ -0,0 +1,106 @@
+package org.bitcoins.scripts
+
+import akka.NotUsed
+import akka.stream.scaladsl.{Keep, Sink, Source}
+import org.bitcoins.core.protocol.blockchain.Block
+import org.bitcoins.core.protocol.transaction.{
+  BaseTransaction,
+  EmptyTransaction,
+  WitnessTransaction
+}
+import org.bitcoins.rpc.client.common.BitcoindRpcClient
+import org.bitcoins.server.BitcoindRpcAppConfig
+import org.bitcoins.server.routes.BitcoinSRunner
+
+import scala.concurrent.Future
+
+/** Useful script for scanning bitcoind
+  * This file assumes you have pre-configured the connection
+  * between bitcoin-s and bitcoind inside of bitcoin-s.conf
+  * @see https://bitcoin-s.org/docs/config/configuration#example-configuration-file
+  */
+class ScanBitcoind(override val args: Array[String]) extends BitcoinSRunner {
+  override val actorSystemName = "scan-bitcoind"
+
+  implicit val rpcAppConfig: BitcoindRpcAppConfig =
+    BitcoindRpcAppConfig(datadir, baseConfig)
+
+  override def startup: Future[Unit] = {
+
+    val bitcoind = rpcAppConfig.client
+
+    val startHeight = 675000
+    val endHeightF: Future[Int] = bitcoind.getBlockCount
+
+    for {
+      endHeight <- endHeightF
+      _ <- countSegwitTxs(bitcoind, startHeight, endHeight)
+      _ <- system.terminate()
+    } yield {
+      sys.exit(0)
+    }
+
+  }
+
+  /** Searches a given Source[Int] that represents block heights applying f to them and returning a Seq[T] with the results */
+  def searchBlocks[T](
+      bitcoind: BitcoindRpcClient,
+      source: Source[Int, NotUsed],
+      f: Block => T,
+      numParallelism: Int =
+        Runtime.getRuntime.availableProcessors() * 2): Future[Seq[T]] = {
+    source
+      .mapAsync(parallelism = numParallelism) { height =>
+        bitcoind
+          .getBlockHash(height)
+          .flatMap(h => bitcoind.getBlockRaw(h))
+          .map(b => (b, height))
+      }
+      .mapAsync(numParallelism) { case (block, height) =>
+        logger.info(
+          s"Searching block at height=$height hashBE=${block.blockHeader.hashBE.hex}")
+        Future {
+          f(block)
+        }
+      }
+      .toMat(Sink.seq)(Keep.right)
+      .run()
+  }
+
+  def countSegwitTxs(
+      bitcoind: BitcoindRpcClient,
+      startHeight: Int,
+      endHeight: Int): Future[Unit] = {
+    val startTime = System.currentTimeMillis()
+    val source: Source[Int, NotUsed] = Source(startHeight.to(endHeight))
+
+    //in this simple example, we are going to count the number of witness transactions
+    val countSegwitTxs: Block => Int = { block: Block =>
+      var counter = 0
+      block.transactions.map {
+        case _: WitnessTransaction =>
+          counter = counter + 1
+        case _: BaseTransaction | EmptyTransaction =>
+        //do nothing
+      }
+      counter
```

Comment:
I think you can do `block.transactions.count(_.isInstanceOf[WitnessTransaction])`